core/orchestrator.go

```diff
@@ -316,6 +320,10 @@ func (orch *orchestrator) acceptablePrice(sender ethcommon.Address, ep *net.Pric
 	return nil
 }
 
+func (orch *orchestrator) isActive() bool {
+	return orch.node.OrchestratorPool.Size() > 0
```

Comment:
Regarding (1) I do remember you said it _could_ be nice to let O have access to `OrchestratorPool` (in on-chain mode) as well in the future. This would be a step in that direction while not introducing unnecessary inefficiencies yet. 

Regarding (2) , didn't think of that, that's true. In hindsight I should have probably covered this with a unit test.

I think it makes sense to write the last initialized round to the DB and use the address filter

Reply:
fixed in 53fd42b6

Reply:
>  do remember you said it could be nice to let O have access to OrchestratorPool (in on-chain mode) as well in the future. This would be a step in that direction while not introducing unnecessary inefficiencies yet.

Hmm IIRC we were talking about letting O have access to the state of the full active set (i.e. keeping O's DB updated with all active set updates) - this would be the case as long as we continue to update the DB for active set updates regardless of what mode the node is running in.

> I think it makes sense to write the last initialized round to the DB

Thinking about this a bit more, I am a bit wary about switching from reading from disk to reading from memory for the last initialized round since the value is read quite often when O receives a segment/payment:

- Used to validate the sender for every ticket in a payment
- Used to check the creation round for every ticket in a payment
- Used to check if O is active

Since these checks are on the critical path for transcoding, I'd actually lean towards continuing to read the value from memory at the cost of passing around an additional type.